### PR TITLE
Add proseWrap: never option to prettier md parsing 

### LINKED ISSUE #21707: chore(docs): Add proseWrap: never option to prettier markdown parsing
## Description

Add `proseWrap: never` to the prettier config options for markdown files in the `docs/` directory (or all files if so inclined). Also add it to the prettier config of [gatsby-i18n-source](https://github.com/gatsbyjs/gatsby-i18n-source/blob/master/.prettierrc.js)

## Motivation

The [Translation Style Guide](https://www.gatsbyjs.org/contributing/translation/style-guide/#use-soft-line-wraps) asks to use soft line wraps instead of hard ones. This makes it easier to track changes across repos without causing weird breaks among split-line sentences.

By adding a lint rule to our documentation files we can ensure that this is always the case.

### PR DESCRIPTION
**Pull request by kartik918**
Fixes #21707

### CODE REVIEW

**Comment by kartik918** [comment]
Oh, thanks ! I'll update it in a bit.

**Comment by kartik918** [comment]
I ran `yarn run format` as you said but the linting tests still seem to be failing @vladar. 
Now I used `prettier --write .` as the test suggests but I want to run all these tests locally first if I could to avoid another failing commit. 
Is there a way to test it like CLI before pushing?
Also if you could help in any other way too, I would appreciate it. Thanks!

**Comment by vladar** [comment]
@kartik918 Mind resolving conflicts? Also, you should probably merge master and run `yarn` in the root of the repo. We recently switched to Prettier 2.0 (#22520), so `yarn` is needed to update Prettier version locally.

As for linting in CI - it runs the following commands:

```
yarn lint:code
yarn lint:other
yarn typecheck
yarn check-repo-fields
```

Currently, it fails on `yarn lint:other` step. You should be able to run them locally if you need.

**Comment by kartik918** [comment]
Thanks for getting back on this, I did everything and have ran the tests locally successfully. They should be all good now.

**Comment by vladar** [comment]
`lint` still fails 🤷 

Did you run `yarn` in the repo root after merging master? You need this to update prettier to the latest version.

**Comment by kartik918** [comment]
Okay I did everything including ``yarn`` at the root to update dependencies. All 4 tests pass locally. I thought the tests were failing because of newer commits and it is still the same.

#### Screenshots:

``yarn lint:code``

test1

``yarn lint:other``

test2

Sorry for the wall of commits haha, please squash and merge if the tests go green some day.

**Comment by kartik918** [comment]
More Screenshots for other tests,

``yarn check-repo-fields``

test3

``yarn typecheck``

test4

This is a weird error since gatsby-core-utils should be visible to babel-preset-gatsby.
@vladar this might be causing it because everything else is ideally green, at least locally.

File: packages/babel-plugin-remove-graphql-queries/package.json [hunk 1]
```diff
@@ -5,9 +5,9 @@
   "repository": {
     "type": "git",
     "url": "https://github.com/gatsbyjs/gatsby.git",
-    "directory": "packages/babel-plugin-remove-graphql-queries"
+    "directory": "packages\\babel-plugin-remove-graphql-queries"
   },
-  "homepage": "https://github.com/gatsbyjs/gatsby/tree/master/packages/babel-plugin-remove-graphql-queries#readme",
+  "homepage": "https://github.com/gatsbyjs/gatsby/tree/master/packages\\babel-plugin-remove-graphql-queries#readme",
```
Comment (vladar) [inline packages/babel-plugin-remove-graphql-queries/package.json:10]: Why do we have this here? Basically every affected `package.json` contains similar change and I suspect that's what breaks things at `yarn check-repo-fields`.

File: packages/babel-plugin-remove-graphql-queries/package.json [hunk 2]
```diff
@@ -5,9 +5,9 @@
   "repository": {
     "type": "git",
     "url": "https://github.com/gatsbyjs/gatsby.git",
-    "directory": "packages/babel-plugin-remove-graphql-queries"
+    "directory": "packages\\babel-plugin-remove-graphql-queries"
```
Comment (vladar) [inline packages/babel-plugin-remove-graphql-queries/package.json:8]: And here

File: packages/babel-plugin-remove-graphql-queries/package.json (hunk 1 above)
Comment (kartik918) [reply packages/babel-plugin-remove-graphql-queries/package.json:10]: I think running prettier did it because I haven't touched these manually. Thanks !

**Comment by kartik918** [comment]
Still getting this error after fixing referencing

``Cannot find module 'gatsby-core-utils'.``

**Comment by kartik918** [comment]
Fixed it, getting a new error. Isn't related to linting though "``Error: Not implemented: navigation``"

**Comment by pvdz** [comment]
Considering the core change here is a one-liner, can I suggest you submit a new PR based on the current master? Just the two commits; prettier bump and the commit with all the prettier-changed changes. The change set and commits are getting a little unwieldy.

The change to gatsby/.prettierrc affects 226 files locally for me. Only `.md` files are affected.

I spot checked most changes and they look okay. Two things jumped out to me:

Code example that seems to get malformed:

```diff
diff --git a/docs/tutorial/writing-documentation-with-docz.md b/docs/tutorial/writing-documentation-with-docz.md
index 5546e1300..535ae15b4 100644
--- a/docs/tutorial/writing-documentation-with-docz.md
+++ b/docs/tutorial/writing-documentation-with-docz.md
@@ -123,10 +123,7 @@ name: Button
 menu: Components
 ---
 
-// highlight-start
-import { Playground, Props } from "docz"
-import { Button } from "./button"
-// highlight-end
+// highlight-start import { Playground, Props } from "docz" import { Button } from "./button" // highlight-end
```

That seems weird, a specific markdown format that's not taken into account by Prettier?

Another one that I'm not 100% sure about, but might be fine (there's a few of these), is
```diff
diff --git a/docs/docs/starters.md b/docs/docs/starters.md
index a9402f1ce..19e551b9e 100644
--- a/docs/docs/starters.md
+++ b/docs/docs/starters.md
@@ -63,7 +63,7 @@ gatsby new blog ./Code/my-local-starter
 Official starters are maintained by Gatsby.
 
 | Starter | Demo/Docs | Use case | Features |
-| -------------------------------------------------------------------------------------------- | ------------------------------------------------------------ | ------------------------------ | ---------------------------- |
+| --- | --- | --- | --- |
 | [gatsby-starter-default](https://github.com/gatsbyjs/gatsby-starter-default) | [Demo](https://gatsby-starter-default-demo.netlify.com/) | Appropriate for most use cases | General Gatsby site |
 | [gatsby-starter-blog](https://github.com/gatsbyjs/gatsby-starter-blog) | [Demo](https://gatsby-starter-blog-demo.netlify.com/) | Create a basic blog | Blog post pages and listings |
 | [gatsby-starter-hello-world](https://github.com/gatsbyjs/gatsby-starter-hello-world) | [Demo](https://gatsby-starter-hello-world-demo.netlify.com/) | Learn Gatsby | Gatsby bare essentials |
```

**Comment by kartik918** [comment]
@pvdz did you run tests after the formatting? If all the checks come back green, let me know and I'll open a new PR for it. 
Also did you run ``yarn run format`` for prettier changes?

Thanks for reviewing this!

**Comment by pvdz** [comment]
I did not run all tests (plus, the only files changed were .md files so I'm not sure any test is even affected by it, apart from linting). I ran `yarn format` to apply Prettier / eslint autofix.

Note that there's some internal discussion around using a different formatter for the docs so I'm going to hold off on this PR/change for a bit until that discussion resolves.

cc @tesseralis

**Comment by pvdz** [comment]
Oh I see this issue was already referenced by that issue :p

**Comment by kartik918** [comment]
I'll still open a new PR with a cleaner commit history since it doesn't seem like ``remark-lint`` will be used anytime soon (also because this trivial PR took too long so I'd be happy if it got merged :). You can decide if you want to merge it.

**Pull request closed without merging**